Calendar circles and Details of each day

TaseskiCS opened the pull request:
![Image 2024-03-20 at 5 49 PM](https://github.com/LaurierCS/POD1/assets/145490990/c208f4fc-6914-4d04-909f-a95f7f5ba46a)
![Image 2024-03-20 at 5 50 PM](https://github.com/LaurierCS/POD1/assets/145490990/dc2de4c8-ab2e-4d26-8c1c-3da91dde3008)
Each circle represents a day exhibited in the first photo, and when the circle/day is pressed, the "day_details" page is shown (second photo). The calculation of each circle/days is done by the Intl library, which accounts for leap years, amount of days in each month, etc.. The details page represents all the "details" aka entries from that day, which I am working on now with the Hive database to fetch and display the data according to each day.

Commit:
added calander page

TaseskiCS committed:
remove commented code

Commit:
rearranged files

Commit:
file rearranged

Commit:
Merge branch 'tony/calander-page' of https://github.com/LaurierCS/POD1 into tony/calander-page

Commit:
Merge branch 'main' into tony/calander-page

Commit:
Merge branch 'main' into tony/calander-page

pabloj2001 commented on mobile/voice_journal_app/lib/calander.dart:
```diff
@@ -0,0 +1,110 @@
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Color.fromRGBO(252, 252, 252, 1),
```
Let's use the colors defined in [theme.dart](https://github.com/LaurierCS/POD1/blob/main/mobile/voice_journal_app/lib/theme.dart), perhaps `lightGray`? This will be good for consistency across the app.

pabloj2001 replied:
Let's do the same for the other colors on this page. Feel free to add new colors to the `theme.dart` file if you need to.

pabloj2001 reviewed (changes requested):
Fantastic work! Left a small comment

Commit:
import fix

Commit:
fetch from database (no testing done)

pabloj2001 committed:
Merge branch 'main' into tony/calander-page

pabloj2001 committed:
Revert main changes

pabloj2001 commented on mobile/voice_journal_app/lib/calander.dart:
```diff
@@ -0,0 +1,121 @@
+      home: const CalendarPage(),
+    );
+  }
+}
```
We only want to have one starting point (`main` function) for the app, which we already have in `main.dart`, so let's remove this and the `CalendarApp` class since we don't need it anymore, and we'll call `CalendarPage` from there instead after this PR is merged.

pabloj2001 commented on mobile/voice_journal_app/lib/day_details.dart:
```diff
@@ -0,0 +1,87 @@
+  }
+
+  Future<List<Recording>> _getRecordings() async {
+    await Hive.initFlutter();
```
Hive already gets initialized in `main.dart` so we don't need to do it again here.

pabloj2001 commented on mobile/voice_journal_app/lib/day_details.dart:
```diff
@@ -0,0 +1,87 @@
+
+  Future<List<Recording>> _getRecordings() async {
+    await Hive.initFlutter();
+    final box = await Hive.openBox<Recording>('recordings');
```
let's use the `Hive.box` function instead since we've had issues with `Hive.openBox`
```suggestion
    final box = Hive.box<Recording>('recordings');
```

pabloj2001 reviewed (changes requested):
Great job! Left some minor comments that need addressing before we merge this PR. I also pulled changes from main into your branch and fixed the merge conflicts that came up. Remember to pull from main often so your branch is up to date!

Alec-Hallman committed:
Pablo suggestions

- removed the open box call
- Removed the run app and calendar page class
- Removed initFlutter hive line

Alec-Hallman committed:
Calendar QOL

Added the recordings as a button to the dates so they can then be opened in the playback page.
Calendar page can now be opened in main

pabloj2001 committed:
Merge branch 'main' into tony/calander-page

pabloj2001 committed:
Integrated calendar into app
